pkg/ctl/install/install.go

```diff
@@ -0,0 +1,435 @@
+				fi.opts.templateParams.GitURL,
+				cloneDir)
+		}
+	}()
+
+	logger.Info("Writing Flux manifests")
+	fluxManifestDir := filepath.Join(cloneDir, fi.opts.gitFluxPath)
+	if err := writeFluxManifests(fluxManifestDir, manifests); err != nil {
+		return err
+	}
+
+	if fi.opts.amend {
+		logger.Info("Stopping to amend the the Flux manifests, please exit the shell when done.")
+		if err := runShell(fluxManifestDir); err != nil {
+			return err
+		}
+	}
+
+	// If we need to create a Namespace, we do it synchronously before invoking kubectl apply.
+	// Otherwise there is a race between the creation of the namespace and the other resources living in it
+	if _, ok := manifests[namespaceFileName]; ok {
+		logger.Info("Creating namespace %s", fi.opts.templateParams.Namespace)
+		if err:= createNamespaceSynchronously(fi.k8sClientSet, fi.opts.templateParams.Namespace); err != nil {
+			return err
+		}
+	}
+
+	logger.Info("Installing Flux into the cluster")
+	if err := fi.applyManifestsInDir(ctx, fluxManifestDir); err != nil {
+		return err
+	}
+
+	logger.Info("Waiting for Flux to start")
+	fluxSSHKey, err :=  waitForFluxToStart(ctx, fi.opts, fi.k8sRestConfig, fi.k8sClientSet)
+	if err != nil {
+		return err
+	}
+	logger.Info("Flux started successfully")
+
+	logger.Info("Committing and pushing Flux manifests to %s", fi.opts.templateParams.GitURL)
+	if err := fi.addCommitAndPushFluxManifests(ctx, cloneDir); err != nil {
+		return err
+	}
+	cleanCloneDir = true
+
+	logger.Info("In order for Flux to operate properly make sure it has SSH access to %s",
+		fi.opts.templateParams.GitURL)
+	logger.Info("Flux's current Public SSH key is:\n%s", fluxSSHKey.Key)
+
+	return nil
+}
+
+func (fi *fluxInstaller) cloneRepo(ctx context.Context) (string, error){
+	cloneDir, err := ioutil.TempDir(os.TempDir(), "eksctl-install-flux-clone")
+	if err != nil {
+		return "", fmt.Errorf("cannot create temporary directory: %s", err)
+	}
+	cloneCtx, cloneCtxCancel := context.WithTimeout(ctx, fi.opts.timeout)
+	defer cloneCtxCancel()
+	args := []string{"clone", "-b", fi.opts.templateParams.GitBranch, fi.opts.templateParams.GitURL, cloneDir}
+	err = runGitCmd(cloneCtx,cloneDir, args...)
+	return cloneDir, err
+}
+
+func (fi *fluxInstaller) addCommitAndPushFluxManifests(ctx context.Context, cloneDir string) error {
+	// Add
+	addCtx, addCtxCancel := context.WithTimeout(ctx, fi.opts.timeout)
+	defer addCtxCancel()
+	if err := runGitCmd(addCtx, cloneDir, "add", "--", fi.opts.gitFluxPath); err != nil {
+		return err
+	}
+
+	// Confirm there is something to commit, otherwise move on
+	diffCtx, diffCtxCancel := context.WithTimeout(ctx, fi.opts.timeout)
+	defer diffCtxCancel()
+	if err := runGitCmd(diffCtx, cloneDir, "diff", "--cached", "--quiet", "--", fi.opts.gitFluxPath); err == nil {
+		logger.Info("Nothing to commit (the repository contained identical manifests), moving on")
+		return nil
+	} else if  _, ok := err.(*exec.ExitError); !ok{
+		return err
+	}
+
+	// Commit
```

Comment:
Same as above... If you are to say what is being done, you could at least be more specific and construct a sentence, but even then - if you aren't doing anything unusual, to me there is no need for a comment.

Reply:
Changed to `gid commit` for, hopefully, more clarity. I still find these helpful, given smaller methods for the individual operations wouldn't make much sense because of Go's verbosity.